sklearn/linear_model/_ridge.py

```diff
@@ -2234,10 +2293,32 @@ def fit(self, X, y, sample_weight=None):
 
         return self
 
+    def get_metadata_routing(self):
+        """Get metadata routing of this object.
 
-class RidgeCV(
-    _RoutingNotSupportedMixin, MultiOutputMixin, RegressorMixin, _BaseRidgeCV
-):
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        .. versionadded:: 1.4
+
+        Returns
+        -------
+        routing : MetadataRouter
+            A :class:`~sklearn.utils.metadata_routing.MetadataRouter` encapsulating
+            routing information.
+        """
+        router = (
+            MetadataRouter(owner=self.__class__.__name__)
+            .add_self_request(self)
+            .add(
+                scorer=check_scoring(self, scoring=self.scoring, allow_none=True),
```

Comment:
From what I understand if we define `self._get_scorer` in `_BaseRidgeCV` and then use it in both the `get_metadata_routing` method and when passing this to `_RidgeGCV` it results in errors when `self.scoring` should be None within `_RidgeGCV`. This is because `check_scoring` actually returns a `_PassthroughScorer` inside `_BaseRidgeCV` and not None.

Reply:
From what I understood `self._get_scorer` should simply return `check_scoring` and then we can instead use the `self._get_scorer` method and this should be defined in `_BaseRidgeCV`. Is this correct?

Reply:
Yes, that sounds about right.

Reply:
Kindly refer to this https://github.com/scikit-learn/scikit-learn/blob/36f6734789fc7e4940792c1cfb6a6e90dfcae484/sklearn/linear_model/_ridge.py#L2027-L2051
`_RidgeGCV` specifically has a branch for handling the case when scorer is None.
`_BaseRidgeCV` on the other hand does not return simply None because of the condition in the `check_scoring` method.

Reply:
I think instead of having the score calculation code inside `fit`, `_RidgeGCV` should have a `score` function which does that calculation, and then `check_scoring` will return that and then we won't need the `if error` branch, do we?

Reply:
I would need to see the implementation to see if it works.

Reply:
From what I can see currently, I don't think it seems convenient to separate this out. Based on the code I posted above we have the following
```python
G_inverse_diag, c = solve(float(alpha), y, sqrt_sw, X_mean, *decomposition) 
```
If we want to handle both the scoring being None as well there being as actual scoring inside the `score` method, we would need to pass `G_inverse_diag` and `c` as parameters to the `score` method which I think would make this a bit different than what our structure for the `score` method currently is.

Reply:
something along the lines of we should have a `self._score` which takes the required args, and call that instead. We should be removing scoring calculation from fit.